tests/rest_api/test_configurations.py

```diff
@@ -653,26 +656,220 @@ def test_resource_hints():
+                            "resource": "users",
+                        },
+                        "group_id": {
+                            "type": "resolve",
+                            "field": "id",
+                            "resource": "groups",
+                        },
+                    },
+                },
+            },
+        ],
+    }
+
+    with pytest.raises(ValueError) as e:
+        rest_api_resources(config)
+
+    error_part_1 = re.escape(
+        "Multiple resolved params for resource user_details: [ResolvedParam(param_name='user_id'"
+    )
+    error_part_2 = re.escape("ResolvedParam(param_name='group_id'")
+    assert e.match(error_part_1)
+    assert e.match(error_part_2)
+
+
+def test_resource_dependent_dependent():
+    config: RESTAPIConfig = {
+        "client": {
+            "base_url": "https://api.example.com",
+        },
+        "resources": [
+            "locations",
+            {
+                "name": "location_details",
+                "endpoint": {
+                    "path": "location/{location_id}",
+                    "params": {
+                        "location_id": {
+                            "type": "resolve",
+                            "field": "id",
+                            "resource": "locations",
+                        },
+                    },
+                },
+            },
+            {
+                "name": "meetings",
+                "endpoint": {
+                    "path": "/meetings/{room_id}",
+                    "params": {
+                        "room_id": {
+                            "type": "resolve",
+                            "field": "room_id",
+                            "resource": "location_details",
+                        },
+                    },
+                },
+            },
+        ],
+    }
+
     resources = rest_api_resources(config)
-    assert resources[0].name == "posts"
-    assert resources[0].table_name == "a_table"
-    assert resources[0].max_table_nesting == 2
-    assert resources[0].write_disposition == "merge"
-    assert resources[0].columns == {"a_text": {"name": "a_text", "data_type": "text"}}
-    schema = resources[0].compute_table_schema()
-    primary_keys = [
-        spec["name"]
-        for _, spec in schema["columns"].items()
-        if spec.get("primary_key", False)
-    ]
-    assert primary_keys == ["a_pk"]
-    merge_keys = [
-        spec["name"]
-        for _, spec in schema["columns"].items()
-        if spec.get("merge_key", False)
-    ]
-    assert merge_keys == ["a_merge_key"]
-    assert resources[0].schema_contract == {"tables": "evolve"}
-    assert schema.get("table_format") == "iceberg"
-    assert resources[0].selected is False
-    # TODO: test if it is parallelized and has spec
+    resource_names = [resource.name for resource in resources]
+    assert ["location_details", "locations", "meetings"] == sorted(resource_names)
```

Comment:
I could not find a way to test that we have parent -> child -> child. 

I tried playing with `resource._get_parent_pipe()` but no success

```
(Pdb) resources[0].name
'locations'
(Pdb) resources[1].name
'location_details'
(Pdb) p resources[1]._get_parent_pipe("foo", resources[0])
Pipe locations [steps: 2] at 5486406096
(Pdb) p resources[1]._get_parent_pipe("foo", resources[1])
Pipe location_details [steps: 2] at 5401533648 data bound to Pipe locations [steps: 2] at 5486406096
```

Do you know of a way to assert that a resource has a specific parent?

Reply:
Try `resources[0]._pipe.parent.name`, and check from the innermost child resource (e.g. "meetings")

Reply:
great, this is what I was looking for. The test is much better now!